FoodTracker/Controllers/MealTableViewController.swift

```diff
@@ -0,0 +1,163 @@
+        case "showDetail":
+            guard let mealDetailViewController = segue.destination as? MealViewController else{
+                fatalError("Unexpected Controller: \(segue.destination)")
+            }
+            guard let mealTableViewCell = sender as? MealTableViewCell else{
+                fatalError("Unexpected Cell: \(sender ?? "")")
+            }
+            guard let indexPath = tableView.indexPath(for: mealTableViewCell) else{
+                fatalError("Selected cell is not in table.")
+            }
+            
+            let selectedMeal = meals[indexPath.row]
+            mealDetailViewController.meal = selectedMeal
+        default:
+            fatalError("Unexpected Segue Identifier; \(segue.identifier ?? "")")
+        }
+    }
+    
+    @IBAction func unwindToMealList(sender: UIStoryboardSegue){
+        
+        if let sourceViewController = sender.source as? MealViewController, let meal = sourceViewController.meal{
+            if let selectedIndexPath = tableView.indexPathForSelectedRow {
+                meals[selectedIndexPath.row] = meal
+                tableView.reloadRows(at: [selectedIndexPath], with: .none)
+            }else{
+                let NewIndexPath = IndexPath(row: meals.count, section: 0)
+                
+                meals.append(meal)
+                tableView.insertRows(at: [NewIndexPath], with: .automatic)
+            }
+            
+            saveMeals()
+        }
+    }
+    
+    //MARK: -Private Methods
+    private func loadData(){
+        
+        if let savedMeals = getMeals(){
+            meals += savedMeals
+        }else{
+            loadSampleMeals()
+        }
+    }
+    
+    private func loadSampleMeals(){
+        
+        let photo1 = UIImage(named: "meal1")
+        let photo2 = UIImage(named: "meal2")
+        let photo3 = UIImage(named: "meal3")
+        
+        guard let meal1 = Meal(mealName: "Fried Fish", mealRating: 4, mealImage: photo1) else {
+            fatalError("Unable to initialize Meal1")
+        }
+        guard let meal2 = Meal(mealName: "Egg Fried Rice", mealRating: 3, mealImage: photo2) else {
+            fatalError("Unable to initialize Meal1")
+        }
+        guard let meal3 = Meal(mealName: "Long Potato Chips", mealRating: 5, mealImage: photo3) else {
+            fatalError("Unable to initialize Meal1")
+        }
+        
+        meals += [meal1, meal2, meal3]
+        
+    }
+
+    private func getMeals() -> [Meal]?{
+        
+        return NSKeyedUnarchiver.unarchiveObject(withFile: Meal.ArchiveUrl.path) as? [Meal]
+    }
+    
+    public func saveMeals(){
+        
+        let isSuccessfulSave = NSKeyedArchiver.archiveRootObject(meals, toFile: Meal.ArchiveUrl.path)
+        
+        if isSuccessfulSave {
+            print("Meals saved successfully.")
+        }else{
+            print("Failed to save data.")
+        }
+    }
+}
+
+//MARK: -Extensions
```

Comment:
The mark should always specify what type of methods it has.
Like there you have written UITableViewDelegate Methods in this extension. So you can name its mark something like: `UITableViewDelegate Methods`